_posts/2024-04-30-A-comprehensive-guide-to-setup-and-connect-self-managed-dashboards-with-Amazon-OpenSearch-domain.md

```diff
@@ -0,0 +1,276 @@
+server.host: "0.0.0.0"
+opensearch.hosts: http://localhost:9200
+```
+2. Create a new Dockerfile like below
+```
+FROM opensearchproject/opensearch-dashboards:2.5.0
+RUN /usr/share/opensearch-dashboards/bin/opensearch-dashboards-plugin remove securityDashboards
+COPY --chown=opensearch-dashboards:opensearch-dashboards opensearch_dashboards.yml /usr/share/opensearch-dashboards/config/
+```
+3. Run this command `docker build --tag=opensearch-dashboards-no-security .` to build a new Docker image with security plugin removed.
+4. Validate if the new image is created by running the `docker images` command
+5. In the below sample `docker-compose.yml`, change the dashboards' image name from `opensearchproject/opensearch-dashboards:2.5.0` to `opensearch-dashboards-no-security` and remove the username and password fields.
+```yml
+version: '3'
+services:
+  opensearch-dashboards:
+    image: opensearchproject/opensearch-dashboards:2.5.0
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    expose:
+      - "5601"
+    environment:
+      OPENSEARCH_HOSTS: '["https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"]'
+      OPENSEARCH_USERNAME: 'xxx'
+      OPENSEARCH_PASSWORD: 'xxxx'
+    networks:
+      - opensearch-net
+networks:
+  opensearch-net:
+```
+6. The new `docker-compose-no-security.yml` file is now created which should look like the below. Now run the `docker-compose up` command to run the containers with new image. Now customer can access the self-managed dashboards by hitting the EC2 endpoint with port `5601`. By doing so, you can conveniently view and interact with all the saved objects
+```yml
+version: '3'
+services:
+  opensearch-dashboards:
+    image: opensearch-dashboards-no-security
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    expose:
+      - "5601"
+    environment:
+      OPENSEARCH_HOSTS: '["https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"]'
+    networks:
+      - opensearch-net
+networks:
+  opensearch-net:
+```
+# 
+# Guide to setup self-managed dashboards in ECS & in EC2 hosted docker container - HTTP basic authentication
+
+## Prerequisite
+AWS managed OpenSearch domain must incorporate Fine-Grained Access Control (FGAC) with HTTP basic authentication, ensuring that a master user is created in the internal user database.
+[Reference here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac-http-auth.html)
+
+## Steps to spin up a self-managed dashboards in ECS
+1. Create a task within ECS Fargate using the dashboards docker image.
+2. When creating a task, under "container definition" in "port mapping" make sure the container ports `5601` and `9200` are added.
+3. Under environment variables, add the mandatory keys and values mentioned in this [doc](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/dashboards.html#dashboards-local) to seamlessly connect with managed service domain.
```

Comment:
Should "environment variables" be either bold or in code font?

Reply:
Lets do BOLD